Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -310,7 +310,11 @@ function Publish-UpgradedServiceFabricApplication
         catch
         {
             Write-Host (Get-VstsLocString -Key SFSDK_UnregisterAppTypeOnUpgradeFailure -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            Unregister-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName -ApplicationTypeVersion $names.ApplicationTypeVersion -Force
+            Write-Host -Message $_.Exception.Message
+            if(!$Action.Equals('Upgrade'))
```

Comment:
I think this will never be true. In deploy.ps1 we never set action to 'upgrade'. To fix #5014 we need to first start supporting just 'upgrade' action.

Reply:
@bishal-pdMSFT 
This change is to fix the semantics of the script, and yes it will always be the true as we never set the action to Upgrade from Deploy.ps1.
Also, I favor that Deploy.ps1 should not set just 'upgrade' because as per current design upgrade is allowed only if user updates the app-version and in such case SF allow to upgrade the application once we copy the updated application package to the image store and register the updated application type.

Reply:
Correct. But additionally task should support scenario mentioned in #5014 - which is register once but upgrade multiple times for different publish profiles. Which means that we need to set action to just "Upgrade" while calling Publish-UpgradedServiceFabricApplication.ps1
@NCarlsonMSFT do you agree that we should start supporting this?

Reply:
This will also fix issue #4153

Reply:
I've added an item to our back-log to add support for just Upgrade

Reply:
@NCarlsonMSFT @bishal-pdMSFT 
To summarize: The common ask from both the issues https://github.com/Microsoft/vsts-tasks/issues/4153 and https://github.com/Microsoft/vsts-tasks/issues/5014 is to provide support just Upgrade i.e. allowing users to upgrade their application even when they don't update application-type version.
I've done the code changes regarding the same in the PR https://github.com/Microsoft/vsts-tasks/pull/5810. Please have a look and let me know if there any scenarios where this change might break the expected functionalities.